blacktie/integration-tests/src/test/java/org/jboss/narayana/blacktie/jatmibroker/xatmi/CSControl.java

```diff
@@ -53,10 +54,14 @@ public void tearDown() {
         try {
             if (server != null) {
                 log.debug("destroying server process");
-                server.interrupt();
-                server.getProcess().destroy();
-                log.debug("destroyed server process");
+                try {
+                    int pid = getUnixPID(server.getProcess());
+                    Runtime.getRuntime().exec("kill -INT " + pid).waitFor();
```

Comment:
The test launches a C program which in turn spawns a C process to run a blacktie server. Before the fix the kill signal kills the CSControl process but not the spawned Blacktie server.